cfe_internal/update/update_bins.cf

```diff
@@ -391,9 +414,14 @@ bundle edit_line u_install_script
 pkgname=`pkginfo -d $(const.dollar)1 | awk '{print $(const.dollar)2}'`
 /usr/sbin/pkgrm -n -a $(cfe_internal_update_bins.admin_file) $pkgname
 /usr/sbin/pkgadd -n -a $(cfe_internal_update_bins.admin_file) -d $(const.dollar)1 all
-$(sys.workdir)/bin/cf-execd || true
-exit 0";
-
+while /usr/bin/pgrep cf-agent || /usr/bin/pgrep cf-execd; do
+echo Maniacally trying to kill all cf-execd, and cf-agent processes after package upgrade to ensure only one copy of the new binaries is running.
+/usr/bin/pkill cf-agent
+/usr/bin/pkill cf-execd
+/etc/init.d/cfengine3 stop
+done
```

Comment:
I'd add "sleep 1" before "done" to give some time to the process to actually die.